js/ui/overlay/overlay_position_controller.js

```diff
@@ -0,0 +1,492 @@
+import $ from '../../core/renderer';
+import { isDefined, isString, isEvent, isWindow } from '../../core/utils/type';
+import { extend } from '../../core/utils/extend';
+import positionUtils from '../../animation/position';
+import { resetPosition, move, locate } from '../../animation/translator';
+import { getWindow } from '../../core/utils/window';
+import { originalViewPort, value as viewPort } from '../../core/utils/view_port';
+import { pairToObject } from '../../core/utils/common';
+
+const window = getWindow();
+
+const OVERLAY_POSITION_ALIASES = {
+    'top': { my: 'top center', at: 'top center' },
+    'bottom': { my: 'bottom center', at: 'bottom center' },
+    'right': { my: 'right center', at: 'right center' },
+    'left': { my: 'left center', at: 'left center' },
+    'center': { my: 'center', at: 'center' },
+    'right bottom': { my: 'right bottom', at: 'right bottom' },
+    'right top': { my: 'right top', at: 'right top' },
+    'left bottom': { my: 'left bottom', at: 'left bottom' },
+    'left top': { my: 'left top', at: 'left top' }
+};
+const OVERLAY_DEFAULT_BOUNDARY_OFFSET = { h: 0, v: 0 };
+
+
+const SIDE_BORDER_WIDTH_STYLES = {
+    'left': 'borderLeftWidth',
+    'top': 'borderTopWidth',
+    'right': 'borderRightWidth',
+    'bottom': 'borderBottomWidth'
+};
```

Comment:
Should we extract it? Looks like we already uses something similar there: https://github.com/DevExpress/DevExtreme/blob/21_2/js/renovation/ui/resizable/utils.ts#L14-L19

Reply:
so attentively, thanks!